### ISSUE TITLE
Istanbul showing no coverage on function generator

### ISSUE THREAD

**Issue by dkreft** [istanbuljs/babel-plugin-istanbul#30]
I'm not sure if this an issue with babel-plugin-istanbul or with istanbul itself, so I'm going to start here.

I've got a really simple file:

``` js
import { fork } from 'redux-saga/effects'

import { pollVesselData } from './vesselSaga'

export default function* Sagas() {
  yield fork(pollVesselData)
}
```

With a simple mocha test file:

``` js
import {
  expect,
  srcRequire,
} from '../testHelper'

import { fork } from 'redux-saga/effects'

const Sagas = srcRequire('sagas')
const { pollVesselData } = srcRequire('sagas/vesselSaga')

describe('Sagas', () => {
  describe('the default saga', () => {
    const fn = Sagas.default()

    it('forks pollVesselData()', function*() {
      expect(fn.next().value).to.eql(fork(pollVesselData))
    })
  })
})
```

The test passes, but the report says 0% of the statements are covered:

```
 src/sagas      |       80 |      100 |       50 |       80 |                |
  index.js      |        0 |      100 |        0 |        0 |              6 |
```

I've got another file in the same directory (`vesselSaga.js`) that has similarly-written tests and it shows 100% coverage.

Here's my `.babelrc`:

``` json
{
  "presets": ["es2015", "react", "stage-2"],
  "plugins": [
    "transform-strict-mode"
  ],
  "env": {
    "test": {
      "plugins": [
        ["istanbul", {
          "exclude": [
            "test/**/*"
          ]
        }]
      ]
    }
  }
}
```

The command that's being executed:

``` sh
$ nyc --reporter text --reporter clover --reporter html npm test
```

And the npm test:

``` sh
$ mocha --compilers js:babel-register --require babel-polyfill --recursive -u bdd-lazy-var/global
```

What am I doing wrong?

**Comment by Row** [comment]
Any update on this? I am having the same problem, using istanbul 1.0.0-alpha.2.

**Comment by wyze** [comment]
Hi @Row, did you mean to have the assertion inside a generator function?

```js
    it('forks pollVesselData()', function*() {
      expect(fn.next().value).to.eql(fork(pollVesselData))
    })
```

When I did this, I noticed my test pass, but it never actually performed the assertion. This explains why the coverage was 0.

When I made it a regular function, then it performed the assertion and my coverage was 100. I'm not sure if `mocha` supports generator functions out of the box, so maybe there is another plugin you have that I am missing.
